### PR TITLE
feat: CP support Drawer closeIcon

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- #46889

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support icons |
| 🇨🇳 Chinese | feat: CP 组件支持 icons  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/config-provider/context.ts [hunk 1]
```diff
@@ -188,6 +188,15 @@ export interface ConfigConsumerProps {
   flex?: FlexConfig;
   wave?: WaveConfig;
   warning?: WarningContextProps;
+  icons?: Partial<
+    Record<
+      | 'closeIcon' // Alert、Drawer、FloatButton、Modal、Notification、Tag、Tour
+      | 'clearIcon' // Cascader、DatePicker、TimePicker、Input、Mentions、Select、TreeSelect
+      | 'suffixIcon' // Cascader、Select、TreeSelect
```
Comment (afc163) [inline components/config-provider/context.ts:195]: suffixIcon 这种语义是模糊的，这样放进来都不知道要替换哪一个图标。我还是建议图标换图标，而非语义指定图标。

Comment (li-jia-nan) [reply components/config-provider/context.ts:195]: `suffixIcon` 语义确实模糊，我先转 WIP，这个下周三开会的时候再讨论一下

**Review by harsh-kp — APPROVED** [review]

File: components/config-provider/context.ts (hunk 1 above)
Comment (li-jia-nan) [reply components/config-provider/context.ts:195]: cc @MadCcc @zombieJ

Comment (zombieJ) [reply components/config-provider/context.ts:195]: 这几个看起来像是 arrowIcon

Comment (MadCcc) [reply components/config-provider/context.ts:195]: 图标换图标应该去 IconContext 做了

Comment (afc163) [reply components/config-provider/context.ts:195]: arrowIcon 也有其他形式，比如 Table 筛选里的上下箭头、InputNumber 里的上下箭头、Tree 的展开收起箭头。

Comment (li-jia-nan) [reply components/config-provider/context.ts:195]: image
image

**Review by vagusX — APPROVED** [review]

File: components/config-provider/context.ts [hunk 2]
```diff
@@ -84,6 +84,7 @@ export interface ButtonConfig extends ComponentStyleConfig {
 export interface DrawerConfig extends ComponentStyleConfig {
   classNames?: DrawerProps['classNames'];
   styles?: DrawerProps['styles'];
+  closeIcon?: DrawerProps['closeIcon'];
```
Comment (MadCcc) [inline components/config-provider/context.ts:87]: 直接 Pick 吧，省的笔误属性对不上

Comment (li-jia-nan) [reply components/config-provider/context.ts]: 其它几个地方也顺便改了：

- 47153

**Review by MadCcc — APPROVED** [review]

**Pull request merged**